sdk/storage/azure-storage-blob/v10_v12.md

```diff
@@ -0,0 +1,148 @@
+# Storage Blob Service SDK Migration Guide from 10.x to 12.x
+
+In this section, we list main changes you need to be aware of when converting your Storage Blob SDK library from Version 10 to Version 12.
+## Prerequisites
+Java Development Kit (JDK) with version 8 or above
+
+## Update reactive framework from RxJava to Reactor
+The biggest payoff here is the switch to Reactor and ReactorNetty. Both Reactor and Reactor-Netty are much more mature than the ReactiveX + custom-ReactiveX-Netty-interface we were using. Both Reactor and ReactorNetty are maintained by Spring and support other large projects, making them safer and more modern, while similar to what was present in v10. For more info, please refer [Reactor Project](https://projectreactor.io/)
+
+## Performance Increase by using ReactorNetty
+Perf increases over purely synchronous HTTP Clients (e.g. OkHttp). In our long running stress tests, we have observed 3x better performance using Reactor-Netty over OkHttp when using our sync apis. The async apis perform even better. Stress tests result. (placeholder)
+
+## Rename client:
+Class name in V10:
+```
+ServiceURL
+ContainerURL
+BlobURL
+BlockBlobURL
+AppendBlobURL
+PageBlobURL
+```
+Sync Class name in V12:
+```
+BlobServiceClient
+ContainerClient
+BlobClient
+BlockBlobClient
+AppendBlobClient
+PageBlobClient
+```
+
+Async Class name in V12:
+```
+BlobServiceAsyncClient
+ContainerAsyncClient
+BlobAsyncClient
+BlockBlobAsyncClient
+AppendBlobAsyncClient
+PageBlobAsyncClient
+```
+
+## Provide sis by side sync-over-async APIs
+Each type (Container, Blob, etc.) has both a sync and async client. It is easy and intuitive to use whichever one is preferable in a given context without having to bother about clutter from the other.
+
+* Example of using Async API in V12. (Suppose the `containerClienthas` been initialized. Please refer the [Build Client](#build-client) section.)
+```
+Mono<Response<BlobProperties>> response = containerClient.getProperties();
+```
+
+* Example of usingSync API in V12
+```
+BlobProperties blobProperties = containerClient.getProperties();
+```
+## Added Maven dependency
+```
+<dependency>
+  <groupId>com.azure</groupId>
+  <artifactId>azure-storage-blob</artifactId>
+  <version>12.0.0-preview.2</version>
+</dependency>
+```
+## Credential
+There are 3 ways of authenticating the client.
+### Connection string 
+In V12, we allow to build connection string through the builder.
+```
+BlockBlobClient client = new BlobClientBuilder()
+              .connectionstring("${connectionstring}").buildBlockBlobClient();
+```
+
+### SAS token
+You can build the SAS token through the client builder `endpoint`. Adds the sasToken from portal at the end of the endpoint. Refer the [documentation](https://github.com/Azure/azure-sdk-for-java/tree/azure-storage-file_12.0.0-preview.3/sdk/storage/azure-storage-blob/README.md) for more info.
```

Comment:
The meaning of build the SAS token is unclear here, are you building a SAS token or building a client with a SAS token credential?